rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -417,7 +423,7 @@ def calculate_spend_cost_basis(
             used_acquisitions=used_acquisitions,
             settings=settings,
             timestamp_to_date=timestamp_to_date,
-            average_cost_basis=self.current_average_cost_basis,
+            average_cost_basis=self.current_total_acb / self.current_shares_amount,
```

Comment:
We need an important note here. According to the formula `self.current_shares_amount` should be the **previous** number of  assets owned.

Which **it is**, but only due to the way that `CostBasisCalculator` calls `calculate_spend_cost_basis` before `reduce_asset_amount` in `spend_asset`.

Can you write that as a comment above, that this needs to hold? Hopefully if we ever break this the tests will let us know